src/Terminal/ui/TerminalRoot.tsx

```diff
@@ -112,20 +112,136 @@ export function TerminalRoot({ terminal, router, player }: IProps): React.ReactE
   }, []);
 
   function lineClass(s: string): string {
-    if (s === "error") {
-      return classes.error;
-    }
-    if (s === "success") {
-      return classes.success;
+    const lineClassMap : Record<string,string> = {
+      "error": classes.error,
+      "success": classes.success,
+      "info": classes.info,
+      "warn": classes.warning,
+    };
+    return lineClassMap[s]||classes.primary;
+  }
+
+  function ansiCodeStyle(code: string|null): Record<string,any> {
+    const COLOR_MAP : Record<string|number,any> = {
+      0: 'black',
+      1: 'red',
+      2: 'green',
+      3: 'yellow',
+      4: 'blue',
+      5: 'magenta',
+      6: 'cyan',
+      7: 'white',
     }
-    if (s === "info") {
-      return classes.info;
+    const ansi2rgb = (code:number):string => {  /* eslint-disable yoda */
+      if((0 <= code) && (code < 8)){
+        // x8 RGB
+        return COLOR_MAP[code]
+      }
+      if((16 <= code) && (code < 232)){
+        // x216 RGB
+        const base = code - 16
+        const ir = Math.floor( base       / 36)
+        const ig = Math.floor((base % 36) /  6)
+        const ib = Math.floor((base %  6) /  1)
+        const r = ir <= 0 ? 0 : 55 + ir * 40
+        const g = ig <= 0 ? 0 : 55 + ig * 40
+        const b = ib <= 0 ? 0 : 55 + ib * 40
+        return `rgb(${r}, ${g}, ${b})`
+      }
+      if((232 <= code) && (code < 256)){
+        // x32 greyscale
+        const base = code - 232
+        const grey = base * 10 + 8
+        return `rgb(${grey}, ${grey}, ${grey})`
+      }
+      // shouldn't get here (under normal circumstances), but just in case
+      return 'initial'
     }
-    if (s === "warn") {
-      return classes.warning;
+    type styleKey = "fontWeight" | "textDecoration" | "color" | "backgroundColor"
+    const style : {
+      fontWeight?: string;
+      textDecoration?: string;
+      color?: string;
+      backgroundColor?: string;
+    } = {}
+
+    if(code !== null){ /* eslint-disable yoda */
```

Comment:
It would be better to return style here if code is null, to reduce indentation as well as makes it clearer exactly what happens if code is null without scrolling to the bottom of this rather large block of code.

```js
if (code === null) {
  return style;
}
```